lib/awful/client.lua

```diff
@@ -900,19 +900,18 @@ function client.setwfact(wfact, c)
     t:emit_signal("property::windowfact")
 end
 
---- Increment a client's window factor
+--- Change a client's window factor.
 --
 -- @legacylayout awful.client.incwfact
--- @param add amount to increase the client's window
+-- @tparam number add Amount to increase/decrease the client's window.
+--   Should be between 0.0 and 1.0.
```

Comment:
The range is [-current_window_factor, infinity]. You don't really want this in the docs, I guess.

The code in this function does the increment and then does normalize(). This function gets all window factors back in the range [0,1].

If the argument to `incwfact` is `-current_factor`, the window ends up with factor zero (and things will break when the layout actually tries to assign the window a size of zero).
If the argument is something huge, all other windows will end up with a factor of zero (and the window we are called on ends up with value 1).
In between these extremes, sane things happen. Further outside of these extremes, breakage ensures (passing in a NaN will permanently  break window factors).